packages/nx/src/utils/plugins/plugin-capabilities.ts

```diff
@@ -39,44 +40,41 @@ export async function getPluginCapabilities(
       pluginName,
       getNxRequirePaths(workspaceRoot)
     );
-    const pluginModule =
-      packageJson.generators ??
-      packageJson.executors ??
-      packageJson['nx-migrations'] ??
-      packageJson['schematics'] ??
-      packageJson['builders']
-        ? await loadNxPluginAsync(
-            pluginName,
-            getNxRequirePaths(workspaceRoot),
-            workspaceRoot
-          )
-        : ({
-            name: pluginName,
-          } as NxPlugin);
+    const pluginModule = await tryGetModule(packageJson, workspaceRoot);
     return {
       name: pluginName,
-      generators:
-        tryGetCollection(
+      generators: {
+        ...tryGetCollection(
           packageJsonPath,
-          packageJson.generators,
-          'generators'
-        ) ||
-        tryGetCollection(
+          packageJson.schematics,
+          'schematics'
+        ),
+        ...tryGetCollection(
           packageJsonPath,
           packageJson.schematics,
           'generators'
```

Comment:
This call and the one below seem to be the same. I think we are missing the case where there's a `generators` entry in the `package.json` and an `schematics` entry in the collection.
```suggestion
          packageJson.generators,
          'schematics'
```